core/src/components/slides/slides.tsx

```diff
@@ -147,15 +148,26 @@ export class Slides implements ComponentInterface {
     }
   }
 
-  async disconnectedCallback() {
+  disconnectedCallback() {
     if (this.mutationO) {
       this.mutationO.disconnect();
       this.mutationO = undefined;
     }
-    const swiper = await this.getSwiper();
-    swiper.destroy(true, true);
-    this.swiper = new Promise(resolve => { this.readySwiper = resolve; });
-    this.swiperReady = false;
+
+    /**
+     * We need to synchronously destroy
+     * swiper otherwise it is possible
+     * that it will be left in a
+     * destroyed state if connectedCallback
+     * is called multiple times
+     */
+    const swiper = this.syncSwiper;
+    if (swiper !== undefined) {
```

Comment:
One potential issue:

In the event that slides is connected and then disconnected before swiper is fully initialized, Swiper will not actually be destroyed here. Need to investigate more, but will garbage collection pick this up for us?